model/components/icinga.yml

```diff
@@ -0,0 +1,31 @@
+---
+components:
+
+  # Icinga 2
+  - id: icinga2
```

Comment:
Would it be better to split into several components? Until now we always at least divided agents from collectors/visualization/.. The reason is that by this way you are saying that your agent can be connected to MySql for example, which is not the case I belive. I guess your agent can be connected to your collector and the collector can be connected to then to other tools. 

The best way usually is to divide into really separate components, meaning one component can run separately from the other. For us, this is also important because of the future features we are planning on the landscape website.

Reply:
I thought about splitting everything up into several component. I would rather split Icinga into separate components, but I thought about the following issues, maybe you can help:

- Does it make sense to split Icinga 2 and Icinga Web 2? The one is the core that runs checks, provides data and stores them into MySQL, PostgreSQL, Graphite and so on. The other is the web interface to display everything. Both of them can be extended with features and modules.
- Icinga 2 runs on the server and also on the agent side. It's the same binary, but depending on it's configuration it can act as a server or agent. I understand the concern. An Icinga agent should not be connected to any database.
- We have even more components, like Icingabeat, Logstash Output for Icinga or several modules for Icinga Web 2 to display data from Graphite or Elasticsearch. Should they be added as separate components as well?
- We don't have logos for all of these components, how should I handle this?

In short, I would add even more components from the Icinga ecosystem, I'm just not sure how to handle it and if it would fit the purpose of this landscape.

Reply:
`Does it make sense to split Icinga 2 and Icinga Web 2?`

I would say yes definitely.

`Icinga 2 runs on the server and also on the agent side. It's the same binary, but depending on it's configuration it can act as a server or agent. I understand the concern. An Icinga agent should not be connected to any database.`

This one is tricky. We did not have such situation until now. From point of being same binary for me this is same component, but as you said then if acting as agent it should not be connected to any storage tools. So here I would also suggest to split.

`We have even more components, like Icingabeat, Logstash Output for Icinga or several modules for Icinga Web 2 to display data from Graphite or Elasticsearch. Should they be added as separate components as well?`

I think this is easy. 
* For the modules don't add any new component, just mark that Icinga Web 2 can be connected to Graphite or Elasticsearch. Modules are just the config you start the already existing component 
* For the logstash ouput, I would also say just connect Icinga server to the Logstash. As this is plugin again for the existing component.
* The beats could be considered as new component imo. I guess this can be in the Transformation category?

`We don't have logos for all of these components, how should I handle this?`

Does not matter, use the same logo everywhere, this is fine. Many tools are already handled like this. However, if any component has it's own logo add it.

As I understand we would end up for 4 components:

* Icinga Agent
* Icinga Server
* Icinga Web
* Icingabeats

Just make sure that connectins, aspects and other properties are correctly divided and defined now.

**Thanks again for the effort, I wanted to do this myself long time ago, but it's always way better that somebody knowing the tool complelty does it.**